lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -947,6 +1413,186 @@ impl From<CGOFileSystemObject> for FileSystemObject {
+    Failed = 1,
+    /// resource does not exist
+    DoesNotExist = 2,
+    /// resource already exists
+    AlreadyExists = 3,
+}
+
+#[derive(Debug, Clone)]
+pub struct SharedDirectoryWriteRequest {
+    completion_id: u32,
+    directory_id: u32,
+    offset: u64,
+    path: String,
+    write_data: Vec<u8>,
+}
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct CGOSharedDirectoryWriteRequest {
+    pub completion_id: u32,
+    pub directory_id: u32,
+    pub offset: u64,
+    pub path_length: u32,
+    pub path: *const c_char,
+    pub write_data_length: u32,
+    pub write_data: *mut u8,
+}
+
+#[derive(Debug)]
+pub struct SharedDirectoryReadRequest {
+    completion_id: u32,
+    directory_id: u32,
+    path: String,
+    offset: u64,
+    length: u32,
+}
+
+#[repr(C)]
+pub struct CGOSharedDirectoryReadRequest {
+    pub completion_id: u32,
+    pub directory_id: u32,
+    pub path_length: u32,
+    pub path: *const c_char,
+    pub offset: u64,
+    pub length: u32,
+}
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct SharedDirectoryReadResponse {
+    pub completion_id: u32,
+    pub err_code: TdpErrCode,
+    pub read_data: Vec<u8>,
+}
+
+impl From<CGOSharedDirectoryReadResponse> for SharedDirectoryReadResponse {
+    fn from(cgo_response: CGOSharedDirectoryReadResponse) -> SharedDirectoryReadResponse {
+        unsafe {
+            SharedDirectoryReadResponse {
+                completion_id: cgo_response.completion_id,
+                err_code: cgo_response.err_code,
+                read_data: from_go_array(cgo_response.read_data, cgo_response.read_data_length),
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct CGOSharedDirectoryReadResponse {
+    pub completion_id: u32,
+    pub err_code: TdpErrCode,
+    pub read_data_length: u32,
+    pub read_data: *mut u8,
+}
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct SharedDirectoryWriteResponse {
+    pub completion_id: u32,
+    pub err_code: TdpErrCode,
+    pub bytes_written: u32,
+}
+
+pub type CGOSharedDirectoryWriteResponse = SharedDirectoryWriteResponse;
+
+/// SharedDirectoryCreateRequest is sent by the TDP server to
+/// the client to request the creation of a new file or directory.
+#[derive(Debug)]
+pub struct SharedDirectoryCreateRequest {
+    completion_id: u32,
+    directory_id: u32,
+    file_type: FileType,
+    path: String,
+}
+
+#[repr(C)]
+pub struct CGOSharedDirectoryCreateRequest {
+    pub completion_id: u32,
+    pub directory_id: u32,
+    pub file_type: FileType,
+    pub path: *const c_char,
+}
+
+/// SharedDirectoryCreateResponse is sent by the TDP client to the server
+/// to acknowledge a SharedDirectoryCreateRequest was received and executed.
+#[derive(Debug, Clone)]
+#[repr(C)]
+pub struct SharedDirectoryCreateResponse {
+    pub completion_id: u32,
+    pub err_code: TdpErrCode,
+}
+
+/// SharedDirectoryListResponse is sent by the TDP client to the server
+/// in response to a SharedDirectoryInfoRequest.
+#[allow(dead_code)]
```

Comment:
Can you remove this or is some of this code still dead?

Reply:
Removed all the unnecessary ones